autorest/validation/validation.go

```diff
@@ -388,8 +388,7 @@ func createError(x reflect.Value, v Constraint, err string) error {
 		v.Target, v.Name, getInterfaceValue(x), err)
 }
 
-// NewErrorWithValidationError appends package type and method name in
-// validation error.
+// NewErrorWithValidationError returns a validation.Error type with the specified arguments.
 func NewErrorWithValidationError(err error, packageType, method string) error {
```

Comment:
>NewErrorWithValidationErro [](start = 5, length = 26)

This name suggests more that the returned error has an inner error a member that is validationError. I know its a breaking change, but maybe it makes sense to call it just `newValidationError` ?

Reply:
Hmm I'm tempted to just remove this function entirely and have the generator emit calls to validation.NewError() instead.  What do you think?